src/main_cache.hpp

```diff
@@ -0,0 +1,19 @@
+#include "base_cache.hpp"
+#include <random>
+
+class MainCache : public BaseCache {
+  public:
+    MainCache(const CACHE_SET &);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Class 'MainCache' has a constructor with 1 argument that is not explicit.](https://app.codacy.com/manual/jerry0715no14/Cache_sim/pullRequest?prid=4735708)